### PR TITLE
Add login to private registry, & run the start agent callback in an async context.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

File: src/ostorlab/scanner/callbacks.py [hunk 1]
```diff
@@ -153,31 +165,73 @@ def _update_state_reporter(
...
-def start_scan(
+def _connect_to_containers_registry(
+    configuration: scanner_conf.RegistryConfig,
+) -> docker.DockerClient:
+    """Connect to container registry."""
+    client = docker.from_env()
+    client.login(
+        username=configuration.username,
+        password=configuration.token,
+        registry=OSTORLAB_REGISTRY,
```
Comment (3asm) [inline src/ostorlab/scanner/callbacks.py:176]: :) this should be coming from the config, not hardcoded here.

Comment (3asm) [reply src/ostorlab/scanner/callbacks.py:176]: Worth doing in another iteration.

File: src/ostorlab/scanner/callbacks.py [hunk 2]
```diff
@@ -153,31 +165,73 @@ def _update_state_reporter(
     return state_reporter
 
 
-def start_scan(
+def _connect_to_containers_registry(
```
Comment (3asm) [inline src/ostorlab/scanner/callbacks.py:168]:
```suggestion
def _connect_containers_registry(
```

Comment (3asm) [reply src/ostorlab/scanner/callbacks.py:168]: Please stop using small words in function names.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/scanner/callbacks.py (hunk 1 above)
Comment (deadly-panda) [reply src/ostorlab/scanner/callbacks.py]: It's a small change, already implemented.

**Review by 3asm — APPROVED** [review]

**Pull request merged**